core/server/common/src/main/java/alluxio/master/AlluxioExecutorService.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master;
+
+import alluxio.concurrent.jsr.ForkJoinPool;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Forwarder over ExecutorService interface for exposing internal queue length.
+ */
+public class AlluxioExecutorService implements ExecutorService {
+  private ExecutorService mExecutor;
+
+  /**
+   * Creates Alluxio ExecutorService wrapper.
+   * 
+   * @param executor underlying executor
+   */
+  public AlluxioExecutorService(ExecutorService executor) {
+    mExecutor = executor;
+  }
+
+  /**
+   * @return the current RPC queue size
+   */
+  public long getRpcQueueLength() {
+    if (mExecutor instanceof ThreadPoolExecutor) {
+      return ((ThreadPoolExecutor) mExecutor).getQueue().size();
+    } else if (mExecutor instanceof ForkJoinPool) {
+      return ((ForkJoinPool) mExecutor).getQueuedSubmissionCount();
+    } else {
+      throw new IllegalArgumentException("Executor is not supported.");
```

Comment:
print the `Executor.getClassName()` here so we know what other types people may want?